--- app.orig.py	2016-09-22 09:03:27.000000000 +0200
+++ app.patch.py	2016-09-22 08:53:57.000000000 +0200
@@ -17,6 +17,7 @@
 
 from oslo_config import cfg
 import oslo_middleware.cors as cors_middleware
+import oslo_middleware.http_proxy_to_wsgi as proxy_middleware
 import pecan
 
 from ironic.api import acl
@@ -94,6 +95,9 @@
         expose_headers=[Version.max_string, Version.min_string, Version.string]
     )
 
+    # Insert the proxy support middleware to generate decent application links.
+    app = proxy_middleware.HTTPProxyToWSGI(app, CONF)
+
     return app
 
 
