--- node_cache.orig.py	2016-12-15 08:53:51.000000000 +0000
+++ node_cache.patch.py	2016-12-15 08:52:37.000000000 +0000
@@ -15,6 +15,7 @@
 
 import copy
 import json
+import six
 import time
 
 from ironicclient import exceptions
@@ -292,7 +293,7 @@
         """
         ironic = ironic or self.ironic
         ports = self.ports()
-        if isinstance(port, str):
+        if isinstance(port, six.string_types):
             port = ports[port]
 
         LOG.debug('Updating port %(mac)s with patches %(patches)s',
@@ -333,7 +334,7 @@
         """
         ironic = ironic or self.ironic
         ports = self.ports()
-        if isinstance(port, str):
+        if isinstance(port, six.string_types):
             port = ports[port]
 
         ironic.port.delete(port.uuid)
@@ -510,8 +511,8 @@
         if not isinstance(value, list):
             value = [value]
 
-        LOG.debug('Trying to use %s of value %s for node look up'
-                  % (name, value))
+        LOG.debug('Trying to use %s of value %s for node look up',
+                  name, value)
         value_list = []
         for v in value:
             value_list.append("name='%s' AND value='%s'" % (name, v))
