--- bigip_init.orig.py	2016-11-23 13:55:49.000000000 +0000
+++ bigip_init.patch.py	2016-11-23 13:56:17.000000000 +0000
@@ -79,7 +79,7 @@
     def _get_tmos_version(self):
         connect = self._meta_data['bigip']._meta_data['icr_session']
         base_uri = self._meta_data['uri'] + 'tm/sys/'
-        response = connect.get(base_uri)
+        response = connect.get(base_uri, verify=False)
         ver = response.json()
         version = urlparse.parse_qs(
             urlparse.urlparse(ver['selfLink']).query)['ver'][0]
