--- l2_service.orig.py	2016-12-15 14:11:55.000000000 +0000
+++ l2_service.patch.py	2016-12-15 14:11:55.000000000 +0000
@@ -130,11 +130,12 @@
 
     def is_common_network(self, network):
         # Does this network belong in the /Common folder?
-        return network['shared'] or \
-            (network['id'] in self.conf.common_network_ids) or \
-            ('router:external' in network and
-             network['router:external'] and
-             self.conf.f5_common_external_networks)
+        # return network['shared'] or \
+        #     (network['id'] in self.conf.common_network_ids) or \
+        #     ('router:external' in network and
+        #      network['router:external'] and
+        #      self.conf.f5_common_external_networks)
+        return True
 
     def get_vlan_name(self, network, hostname):
         # Construct a consistent vlan name
