--- barbican_cert.orig.py	2016-09-22 16:26:55.000000000 +0100
+++ barbican_cert.patch.py	2016-09-22 16:31:38.000000000 +0100
@@ -105,7 +105,7 @@
 
                 # NOTE: Session is deprecated in keystoneclient 2.1.0
                 # and will be removed in a future keystoneclient release.
-                sess = Session(auth=auth)
+                sess = Session(auth=auth, verify=False)
                 self.barbican = Client(session=sess)
 
                 # test barbican service
