--- keystone.orig.py	2016-10-03 19:14:49.000000000 +0100
+++ keystone.patch.py	2016-10-03 19:14:49.000000000 +0100
@@ -105,7 +105,7 @@
 
         try:
             kc = client.Password(**kwargs)
-            _SESSION = session.Session(auth=kc)
+            _SESSION = session.Session(auth=kc,verify=False)
         except Exception:
             with excutils.save_and_reraise_exception():
                 LOG.exception(_LE("Error creating Keystone session."))
