--- neutron/patches/f5/service_builder.orig.py	2016-11-09 14:37:09.000000000 +0000
+++ neutron/patches/f5/service_builder.patch.py	2016-11-09 14:37:09.000000000 +0000
@@ -94,11 +94,10 @@
             segment_data = self.disconnected_service.get_network_segment(
                 context, agent_config, network)
             if segment_data:
-                network['provider:segmentation_id'] = \
-                    segment_data.get('segmentation_id', None)
-                if 'provider:network_type' in network:
-                    network['provider:network_type'] = \
-                        segment_data.get('network_type', None)
+
+                network['provider:segmentation_id'] = segment_data.get('segmentation_id', None)
+                network['provider:network_type'] = segment_data.get('network_type', None)
+                network['provider:physical_network'] = segment_data.get('physical_network', None)
             network_map[network_id] = network
 
             # Check if the tenant can create a loadbalancer on the network.
