--- neutron/patches/lbaas/loadbalancer_dbv2.orig.py	2016-11-10 10:26:41.000000000 +0000
+++ neutron/patches/lbaas/loadbalancer_dbv2.patch.py	2016-11-10 10:27:19.000000000 +0000
@@ -60,7 +60,16 @@
         resource = None
         try:
             if for_update:
-                query = self._model_query(context, model).filter(
+                #query = self._model_query(context, model).filter(
+                #    model.id == id).with_lockmode('update')
+                #resource = query.one()
+
+                # To lock the instance for update, return a single
+                # instance, instead of an instance with LEFT OUTER
+                # JOINs that do not work in PostgreSQL
+                query = self._model_query(context, model).options(
+                    lazyload('*')
+                ).filter(
                     model.id == id).with_lockmode('update')
                 resource = query.one()
             else:
