diff --git a/nova/network/neutronv2/api.py b/nova/network/neutronv2/api.py
index 7050f97211..b70b5ccf8b 100644
--- a/nova/network/neutronv2/api.py
+++ b/nova/network/neutronv2/api.py
@@ -1656,7 +1656,8 @@ class API(base_api.NetworkAPI):
         elif vif_type == network_model.VIF_TYPE_DVS:
             # The name of the DVS port group will contain the neutron
             # network id
-            bridge = port['network_id']
+            bridge = port_details.get(network_model.VIF_DETAILS_BRIDGE_NAME,
+                                      port['network_id'])
         elif (vif_type == network_model.VIF_TYPE_VHOSTUSER and
          port_details.get(network_model.VIF_DETAILS_VHOSTUSER_OVS_PLUG,
                           False)):
