--- vif.orig.py	2016-10-10 09:54:52.000000000 +0100
+++ vif.patch.py	2016-10-10 09:54:52.000000000 +0100
@@ -138,7 +138,10 @@
         if network_id is None:
             # Make use of the original one, in the event that the
             # port binding does not provide this key in VIF details
-            network_id = vif['network']['bridge']
+            #network_id = vif['network']['bridge']
+
+            network_id = (CONF.vmware.integration_bridge or "br-int")
+
         network_ref = network_util.get_network_with_the_name(
                 session, network_id, cluster)
         if not network_ref:
