--- l3_router_hosting_device_scheduler.orig.py	2016-09-01 11:58:52.000000000 +0100
+++ l3_router_hosting_device_scheduler.patch.py	2016-08-17 08:59:34.000000000 +0100
@@ -109,8 +109,8 @@
         query = query.having(
             expr.or_(func.sum(
                 hd_models.SlotAllocation.num_allocated) <= slot_threshold,
-                func.sum(hd_models.SlotAllocation.num_allocated ==
-                         expr.null())))
+                func.sum(hd_models.SlotAllocation.num_allocated) ==
+                         expr.null()))
         query = query.order_by(hd_models.HostingDevice.created_at)
         return query.all()
 
