--- device_status.orig.py	2016-09-01 11:55:29.000000000 +0100
+++ device_status.patch.py	2016-09-01 11:55:29.000000000 +0100
@@ -168,9 +168,14 @@
             return False
 
         # Modifying the 'created_at' to a date time object if it is not
+
+        # *********************
+        # Patch with %.f to avoid exceptions
+        # *********************
+
         if not isinstance(hd['created_at'], datetime.datetime):
             hd['created_at'] = datetime.datetime.strptime(hd['created_at'],
-                                                          '%Y-%m-%d %H:%M:%S')
+                                                          '%Y-%m-%d %H:%M:%S.%f')
 
         if _is_pingable(hd_mgmt_ip):
             LOG.debug("Hosting device: %(hd_id)s@%(ip)s is reachable.",
