--- dhcp.orig.py	2016-08-08 09:50:49.000000000 +0100
+++ dhcp.patch.py	2016-08-08 09:50:49.000000000 +0100
@@ -49,6 +49,16 @@
 DHCPV6_PORT = 547
 METADATA_DEFAULT_PREFIX = 16
 METADATA_DEFAULT_IP = '169.254.169.254'
+
+# **********************************
+# Patch to use METADATA_DEFAULT_CIDR
+# as a proper /16 network fixes
+# Cirros and arguably should go
+# upstream
+# **********************************
+
+METADATA_DEFAULT_NETWORK = '169.254.0.0'
+
 METADATA_DEFAULT_CIDR = '%s/%d' % (METADATA_DEFAULT_IP,
                                    METADATA_DEFAULT_PREFIX)
 METADATA_PORT = 80
@@ -678,7 +688,8 @@
                           (port.mac_address, name, ip_address))
 
         common_utils.replace_file(filename, buf.getvalue())
-        LOG.debug('Done building host file %s', filename)
+        LOG.debug('Done building host file %s with contents:\n%s', filename,
+                  buf.getvalue())
         return filename
 
     def _get_client_id(self, port):
@@ -802,13 +813,24 @@
                 (isolated_subnets[subnet.id] and
                     self.conf.enable_isolated_metadata and
                     subnet.ip_version == 4)):
+
+                # *******************
+                # Change to metadata network
+                # and using dhcp not gateway
+                # *******************
+
                 subnet_dhcp_ip = subnet_to_interface_ip[subnet.id]
                 host_routes.append(
-                    '%s/32,%s' % (METADATA_DEFAULT_IP, subnet_dhcp_ip)
+                    '%s/%s,%s' % (METADATA_DEFAULT_NETWORK,METADATA_DEFAULT_PREFIX, subnet_dhcp_ip)
                 )
             elif not isolated_subnets[subnet.id] and gateway:
+                # *******************
+                # Change to metadata network
+                # and using dhcp not gateway
+                # *******************
+                subnet_dhcp_ip = subnet_to_interface_ip[subnet.id]
                 host_routes.append(
-                    '%s/32,%s' % (METADATA_DEFAULT_IP, gateway)
+                    '%s/%s,%s' % (METADATA_DEFAULT_NETWORK,METADATA_DEFAULT_PREFIX, subnet_dhcp_ip)
                 )
 
             if subnet.ip_version == 4:
